src/sdk/pynni/nni/compression/tensorflow/compressor.py

```diff
@@ -35,25 +37,13 @@ def __init__(self, model, config_list):
         self.modules_to_compress = []
 
     def compress(self):
-        """
-        Compress the model with algorithm implemented by subclass.
-
-        The model will be instrumented and user should never edit it after calling this method.
-        `self.modules_to_compress` records all the to-be-compressed layers
-        """
-        for op in self.bound_model.get_operations():
-            weight_index = _detect_weight_index(op)
-            if weight_index is None:
-                _logger.warning('Failed to detect weight for layer %s', op.name)
-                return
-            weight_op = op.inputs[weight_index].op
-            weight = weight_op.inputs[0]
-
-            layer = LayerInfo(op, weight, weight_op)
+        for keras_layer in self.bound_model.layers:
+            layer = LayerInfo(keras_layer)
             config = self.select_config(layer)
             if config is not None:
                 self._instrument_layer(layer, config)
                 self.modules_to_compress.append((layer, config))
+
```

Comment:
would be better to follow the pytorch version, where we have a member func `detect_modules_to_compress`

Reply:
Thanks, fixed.